src/typeahead/typeahead.directive.ts

```diff
@@ -276,6 +276,10 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
         dropup: this.dropup
       });
 
+    this._outsideClickListener = this.renderer.listen('document', 'click', () => {
```

Comment:
host('document:click') ?

Reply:
It was done this way to achieve better performance. A lot of host listeners will decrease it so I decided to attach listener only when box with results is open.